### DISCUSSION TITLE
Add Cache flexible to Computed Property.

### DISCUSSION THREAD

**Discussion by eznix86** [livewire/livewire#8974] [3. Feature ideas]
It would be nice if we had, the newly created [Cache Flexible](https://laravel.com/docs/11.x/cache#swr) of Laravel to Livewire components.

Where it is useful:
- Whenever you have graphs, or just enough stale time to recompute the cache.

```php
#[Computed(cache: true, flexible: true, window: [15, 30])]
public function orders()
{
    return Metrics::ordersCompleted();
}

// or

#[Computed(persist: true, flexible: true, window: [15, 30])]
public function zombiesKilled()
{
    return Metrics::zombiesKilledByUser(Auth::user());
}
```

Made an implementation:

https://github.com/eznix86/livewire/commit/a558cec28e3086ad13bbd27134f42c651626816d

**Comment by sjwatts119** [comment]
Also interested in this feature

**Reply by lrljoe** [reply]
Think the only stumbling block here, will be that as Livewire v3 supports Laravel 10 (which doesn't have flexible caching), it'd have to be backwards compatible gracefully.

**Reply by eznix86** [reply]
I think you are right, maybe a little if statement
